src/Pelagos/Bundle/AppBundle/Command/BackFillApprovedDateDifCommand.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Command;
+
+use Pelagos\Entity\Dataset;
+use Pelagos\Entity\DIF;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Back fill Script for approved date attribute in DIF entity.
+ *
+ * @see ContainerAwareCommand
+ */
+class BackFillApprovedDateDifCommand extends ContainerAwareCommand
+{
+    /**
+     * Configures the current command.
+     *
+     * @return void
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('dataset:backfill-dif-approved-date')
+            ->setDescription('Backfill DIF approved date for approved datasets.');
+    }
+
+    /**
+     * Executes the current command.
+     *
+     * @param InputInterface  $input  An InputInterface instance.
+     * @param OutputInterface $output An OutputInterface instance.
+     *
+     * @return void
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $entityManager = $this->getContainer()->get('doctrine.orm.entity_manager');
+
+        $datasets = $entityManager->getRepository(Dataset::class)->findBy(array('identifiedStatus' => DIF::STATUS_APPROVED));
```

Comment:
This won't work, we need a historical approval date. So we would have to use the version history with entity Audit.

Reply:
I've used the entity audit bundle 👍